src/git-directory-helper.ts

```diff
@@ -0,0 +1,91 @@
+import * as core from '@actions/core'
```

Comment:
This logic was in git-source-provider but i moved it into a separate file in order to:
1. reduce amount of code in git-source-provider (readability)
2. add lots of tests for this logic (see git-directory-helper.test.ts)

Still the same logic, but different file now. And covered by unit tests now :)